### PR TITLE
Updated chat example to use ncurses

### PR DESCRIPTION
**Pull request by beatgammit**
There are a few hacks, but I don't know ncurses well enough to know if there's a better way of doing it.
#40

### CODE REVIEW

**Comment by jcelliott** [comment]
Awesome!

Found one bug though: the method that gets input needs to have a pointer receiver, otherwise it always returns an empty string.

File: examples/chat/chatclient/chatwin.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	if x == cw.msgRows {
+		x = 0
+		y++
+	} else {
+		x++
+	}
+	return y, x
+}
+
+func (cw chatWin) meh(win *goncurses.Window) (line string) {
```
Comment (jcelliott) [inline examples/chat/chatclient/chatwin.go]: Should be `(cw *chatWin)`. Didn't work for me until I changed that.

Comment (jcelliott) [inline examples/chat/chatclient/chatwin.go]: Also, I can tell you went all-out on coming up with this method name :laughing:

Comment (beatgammit) [inline examples/chat/chatclient/chatwin.go]: Just wanted to see if you'd actually review it... :wink:

I've updated the things with better names and correct `*` usage.

**Pull request merged**